ynr/apps/sopn_parsing/helpers/parse_tables.py

```diff
@@ -165,6 +166,24 @@ def get_description(description, sopn):
     if qs.exists():
         return qs.first()
 
+    if "&" in description:
+        description = description.replace("&", "and")
+        qs = PartyDescription.objects.filter(
+            description__istartswith=description, party__register=register
+        )
+        if qs.exists():
+            return qs.first()
+    else:
+        avg = find_lev(description, register)
+        if avg > 1.2:
+            description = re.findall("[A-Z][^A-Z]*", description)
+            description = (" ".join(description)).strip()
+            qs = PartyDescription.objects.filter(
+                description__istartswith=description, party__register=register
+            )
+            if qs.exists():
+                return qs.first()
+
```

Comment:
I was thinking that maybe these check should be in `get_description` since that's where the errors are coming from. 

I’ve got the levenshtein threshold of two arrays(problematic party descriptions vs their correct descriptions), but I’m struggling to see how to apply this in parse_tables.py . Could it be the description vs the register? 

I also recognise that these conditions are completely unrelated; but wanted to at least capture both bugs in some form to be reworked.

Reply:
When it comes to the "&"/"and" issue, [I have come at this from a different angle in this commit](https://github.com/DemocracyClub/yournextrepresentative/pull/1711/commits/d43140120db9d0145b116e87865af4b51fe98daf). 

Here I am normalising the string we are using to search for in the `clean_description` method (as you had done previously, replacing '&' with 'and') and making use of the django `Replace` function to annotate a new field to the `Party` and `PartyDescription` querysets, where we do the same thing with the name/description field.

Then I am filtering against this new `search_text` field rather than name/description field. From testing this solves the "&"/"and" issue, and the concept could also be extended if there were other similar issues where we wanted to replace a character with something else